pkg/_internal/gracePeriod/gracePeriod.go

```diff
@@ -0,0 +1,57 @@
+package gracePeriod
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/Kuadrant/multicluster-gateway-controller/pkg/_internal/metadata"
+)
+
+const (
+	graceTimestampAnnotation = "kuadrant.io/grace-timeout"
+)
+
+var ErrGracePeriodNotExpired = fmt.Errorf("grace period has not yet expired")
+
+func WasGracePeriodNotExpiredErr(e error) bool {
```

Comment:
can we keep this consistent with the other error checking functions? ```IsGracePeriodNotExpiredErr(err)``` also why not just check it is equal to `ErrGracePeriodNotExpired`?
```
if err == nil{
return false}
return err == ErrGracePeriodNotExpired
```

Reply:
I did initially do it that way, but the error get's wrapped in another error, which is why I changed the name and the matching system to this, rather than reducing the useful info in the error.

Reply:
This still won't work for me, unless I remove some debugging info along the return path.

The current return flow for this error looks like:
-> return ErrGracePeriodNotExpired
-> return fmt.Errorf(err, "extra debugging info here")
-> check if it's a gracePeriodNotExpired error.

Reply:
yeah so I think you need to use the Wrap.

you can use %w I believe
```
err = fmt.Errorf("some info : %w.", ErrGracePeriodNotExpired)
```

Reply:
%w wraps the error and then go can use the Unwrap which is used underneath in `Is`

Reply:
@philbrookes lets change it to use the wrapped error and switch to `IsGracePeriodNotExpiredErr`